webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -0,0 +1,372 @@
+        if (archivedIndex !== -1) {
+          newColumns = [...settings.columns];
+          newColumnWidths = [...settings.columnWidths];
+          newColumns.splice(archivedIndex, 1);
+          newColumnWidths.splice(archivedIndex, 1);
+        } else {
+          newColumns = settings.columns;
+          newColumnWidths = settings.columnWidths;
+        }
+      }
+      updateSettings({
+        archived: showArchived,
+        columns: newColumns,
+        columnWidths: newColumnWidths,
+        row: undefined,
+      });
+    },
+    [settings, updateSettings],
+  );
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetchExperiments();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    return () => {
+      canceler.abort();
+      stopPolling();
+    };
+  }, [canceler, stopPolling]);
+
+  const tabBarContent = useMemo(() => {
+    const getMenuProps = (): DropDownProps['menu'] => {
+      const MenuKey = {
+        Columns: 'columns',
+        ResultFilter: 'resetFilters',
+        SwitchArchived: 'switchArchive',
+      } as const;
+
+      const funcs = {
+        [MenuKey.SwitchArchived]: () => {
+          switchShowArchived(!settings.archived);
+        },
+        [MenuKey.Columns]: () => {
+          handleCustomizeColumnsClick();
+        },
+        [MenuKey.ResultFilter]: () => {
+          resetFilters();
+        },
+      };
+
+      const onItemClick: MenuProps['onClick'] = (e) => {
+        funcs[e.key as ValueOf<typeof MenuKey>]();
+      };
+
+      const menuItems: MenuProps['items'] = [
+        {
+          key: MenuKey.SwitchArchived,
+          label: settings.archived ? 'Hide Archived' : 'Show Archived',
+        },
+        { key: MenuKey.Columns, label: 'Columns' },
+      ];
+      if (filterCount > 0) {
+        menuItems.push({ key: MenuKey.ResultFilter, label: `Clear Filters (${filterCount})` });
+      }
+      return { items: menuItems, onClick: onItemClick };
+    };
+    return (
+      <div className={css.tabOptions}>
+        <Space className={css.actionList}>
+          <Toggle checked={settings.archived} label="Show Archived" onChange={switchShowArchived} />
+          <Button onClick={handleCustomizeColumnsClick}>Columns</Button>
+          <FilterCounter activeFilterCount={filterCount} onReset={resetFilters} />
+        </Space>
+        <div className={css.actionOverflow} title="Open actions menu">
+          <Dropdown menu={getMenuProps()} trigger={['click']}>
+            <div>
+              <Icon name="overflow-vertical" />
+            </div>
+          </Dropdown>
+        </div>
+      </div>
+    );
+  }, [
+    filterCount,
+    handleCustomizeColumnsClick,
+    resetFilters,
+    settings.archived,
+    switchShowArchived,
+  ]);
+
+  const [columnIds, setColumnIds] = useState(settings.columns as string[]);
+  useEffect(() => {
+    // updateSettings({ columns });
+  }, [columns, updateSettings]);
```

Comment:
I didn't wire up to settings for now, I think it makes the most sense for that to be done either after or as a part of bringing the dynamic columns retrieved from the API, since that will pose the biggest challenge for this type of thing.

Reply:
fair enough